creator-node/src/snapbackSM.js

```diff
@@ -47,62 +51,330 @@ class SnapbackSM {
+    if (recoveredSpID === 0) {
+      this.initialized = false
+    }
+    config.set('spID', recoveredSpID)
+    this.log(`Recovered ${config.get('spID')} for ${endpoint}`)
+    this.initialized = true
+    return this.initialized
   }
 
+
   /*
     Initialize the configs necessary to run
   */
   async init () {
-    const endpoint = config.get('creatorNodeEndpoint')
+    await this.stateMachineQueue.empty()
+    await this.syncQueue.empty()
+
     const isUserMetadata = config.get('isUserMetadataNode')
     if (isUserMetadata) {
       this.log(`SnapbackSM disabled for userMetadataNode. ${endpoint}, isUserMetadata=${isUserMetadata}`)
       return
     }
 
+    await this.initializeInternal()
     if (!this.initialized) {
-      this.log(`Initializing SnapbackSM`)
-      this.log(`Retrieving spID for ${endpoint}`)
-      const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
-        endpoint
-      )
-      // A returned spID of 0 means this endpoint is currently not registered on chain
-      // In this case, the stateMachine is considered to be 'uninitialized'
-      if (recoveredSpID === 0) return
-      config.set('spID', recoveredSpID)
-      this.initialized = true
+      return
     }
-    this.log(`Recovered ${config.get('spID')} for ${endpoint}`)
+
 
     // TODO: Enable after dev
     // Run the task every x time interval
     // this.stateMachineQueue.add({}, { repeat: { cron: '0 */x * * *' } })
 
     // Enqueue first state machine operation
     // TODO: Remove this line permanently prior to final check-in
-    // this.stateMachineQueue.add({ startTime: Date.now() })
+    this.stateMachineQueue.add({ startTime: Date.now() })
 
     // Process state machine operations
-    this.stateMachineQueue.process(async (job, done) => {
-      try {
-        await this.processStateMachineOperation(job)
-      } catch (e) {
-        logger.info(`Error processing ${e}`)
-      } finally {
-        this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
-        await utils.timeout(DevDelayInMS)
-        this.stateMachineQueue.add({ startTime: Date.now() })
-        done()
+    this.stateMachineQueue.process(
+      async (job, done) => {
+        try {
+          await this.processStateMachineOperation(job)
+        } catch (e) {
+          this.log(`stateMachineQueue error processing ${e}`)
+        } finally {
+          // TODO: Remove dev mode
+          this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
+          await utils.timeout(DevDelayInMS)
+          this.stateMachineQueue.add({ startTime: Date.now() })
+          done()
+        }
+      }
+    )
+    // Entrypoint to the sync queue, as drained will issue syncs
+    this.syncQueue.process(
+      async (job, done) => {
+        try {
+          await this.processSyncOperation(job)
+        } catch (e) {
```

Comment:
Let us write a mad-dog test to issue parallel updates for >MaxParallelSyncJobs and validate functionality

Reply:
@SidSethi to take this